packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp

```diff
@@ -6009,12 +6013,20 @@ namespace Tpetra {
     // without unpacking the import/export buffer, we don't know how many of the
     // numPacketsPerLID[i] LIDs exist in the target. Below, it is assumed that
     // none do, and padding is requested for all.
-    for (size_t i = 0; i < numEnt; ++i) {
-      auto result = padding.insert (importLIDs_h[i], numPacketsPerLID_h[i]);
+    //
+    // Use tmp_padding since Kokkos::UnorderedMap does not allow re-insertion
+    std::map<local_ordinal_type, size_t> tmp_padding;
+    for (size_t i = 0; i < numEnt; ++i)
+      tmp_padding[importLIDs_h[i]] += numPacketsPerLID_h[i];
+
+    using padding_type = Kokkos::UnorderedMap<local_ordinal_type, size_t, device_type>;
```

Comment:
We're assuming UVM here.  That's fine for now, but if UnorderedMap's constructor initializes its data on device, then we might need to insert a `Kokkos::fence()` after `padding`'s constructor.  Otherwise, this code's correctness may depend on whether `CUDA_LAUNCH_BLOCKING` is set.